### DISCUSSION TITLE
v0.6.0

### DISCUSSION THREAD

**Discussion by arxanas** [arxanas/git-branchless#653] [Release notes]
To install or update `git-branchless`, run the following:

```
$ cargo install --locked git-branchless
$ git branchless init  # in your repository
```

If you were using a remote main branch, you will have to switch to a local main branch, as remote main branches are no longer supported; see below.

If you haven't already taken [the user survey](https://github.com/arxanas/git-branchless/discussions/330), please consider doing so!

## Release highlights

- [Changes to branching](https://github.com/arxanas/git-branchless/discussions/595):
  - We now use "switch" instead of "checkout" as the terminology, and the alias `git sw` is installed by default instead of `git co`. Thanks to @mlcui-google for fixing a bug in this!
  - Remote main branches are no longer supported.
  - Branches will be automatically checked out when switched to. Thanks to @bcspragu for implementing this!
- [More customizable smartlogs](https://github.com/arxanas/git-branchless/discussions/644):
  - You can now render different views of the smartlog, possibly omitting commits and branches at your discretion. Thanks to @claytonrcarter for implementing this!
  - Merge commits are also rendered slightly better.
- [`git test`](https://github.com/arxanas/git-branchless/discussions/643):
  - Lets you run a given command or check on each commit in your stack.
  - You can leave feedback about the UI or workflow in the discussion board or Discord chat.
  - Documentation at [Command: git test](https://github.com/arxanas/git-branchless/wiki/Command:-git-test).

### Added

- (#576) EXPERIMENTAL: Created `git test` command to run a command on each commit in your stack.
- (#589, #593) `git sync --pull` rebases the local main branch on top of the remote main branch.
- (#619) `git smartlog` now renders revsets with non-contiguous commits.
- (#638) `git smartlog` now deduplicates merge commits which appear in the smartlog.

### Changed

- (#526) BREAKING: When checking out a commit with a single branch attached to it, that branch will also be checked out.
- (#589) BREAKING: Remote main branches are no longer supported. To upgrade you'll need to create a local main branch which tracks the remote main branch. You can use `git sync` to keep it up-to-date.
- (#571) Added `current(<revset>)` revset function to determine the latest versions of a set of rewritten commits.
- (#582) Added `main()`, `public()` revset functions.
- (#593, #608) `git sync` produces more descriptive output.

### Fixed

- (#588) `git branchless init` now works when invoked in a worktree.
- (#589) `git reword --force-rewrite` no longer moves remote-tracking branches.
- (#589) `git smartlog` no longer uses 100% CPU in certain cases with remote main branches (because remote main branches are no longer supported).

## New Contributors

Thanks to the following first-time contributors!

* @bcspragu made their first contribution in https://github.com/arxanas/git-branchless/pull/526
* @mlcui-google made their first contribution in https://github.com/arxanas/git-branchless/pull/626

If you're interested in contributing, check out the Developer Guide in the [Wiki](https://github.com/arxanas/git-branchless/wiki), or post in the [Discussions](https://github.com/arxanas/git-branchless/discussions) or [Discord server](https://discord.gg/caYQBJ82A4) to ask any questions.

<hr /><em>This discussion was created from the release <a href='https://github.com/arxanas/git-branchless/releases/tag/v0.6.0'>v0.6.0</a>.</em>

**Comment by jlisee** [comment]
Thanks for the update, I really how fast this can rebase branches.  Any chance on a binary release for this version?

**Reply by arxanas** [reply]
You can find binaries for the upcoming v0.7.0 here: https://github.com/arxanas/git-branchless/releases/tag/v0.7.0-rc.1
